### ISSUE TITLE
Some requests hanging for 1-2 minutes (using HTTPS)

### ISSUE THREAD

**Issue by Moviemoviemoviemovie** [omdbapi/OMDb-API#18]
I'm not sure why this occurs. It will randomly happen with all different kinds of requests (searches, movies, tv). I'll be able to recieve a response for other requests before I am finally able to recieve a response from the initial request which seems to take about 1-2 minutes.

**Comment by josezulu** [comment]
I had the same issue. I guess you are using "https" to access the api, try "http".

It's a workaround, but this issue needs to be solved. It seems to have to do with Couldflare, the author of the api is aware of this, it's been reported by other users too.

**Comment by Moviemoviemoviemovie** [comment]
Oh darn. I can only use "https" right now because I'm calling it from a webpage that only has https. Thanks for the info though.

**Comment by josezulu** [comment]
In that case, contact the author of the api, he might be able to help you so you can still access thru https. He mentioned to me other workarounds.

**Comment by omdbapi** [comment]
@Moviemoviemoviemovie I'll be switching my SSL certs over to CloudFlare next month which should resolve a lot of these issues. 

There are many factors that cause SSL to be slow with OSCP & CLR servers causing additional delays: https://blog.cloudflare.com/how-cloudflare-is-making-ssl-fast/

**Comment by omdbapi** [comment]
Hey @Moviemoviemoviemovie and @josezulu I went ahead and changed my SSL cert over to CloudFlares dedicated CA, please let me know if there is any drop or improvement in performance.

**Comment by josezulu** [comment]
Thanks for the heads up, I'll test in the weekend, keep you posted.

**Comment by trashr0x** [comment]
Hi @omdbapi, unfortunately this hasn't solved it for me yet - some requests will hang as per the OP's description. I'll try plain http and see how that goes.

Update: after some testing, I can confirm that the issue goes away completely when http is used in place of https.

**Comment by josezulu** [comment]
@omdbapi: I confirm I still observe the issue if I use https.
As soon as I started testing, I got the timeout immediately, 1 timeout out of 2 requests. I'm reverting to non secure connection.

**Comment by josezulu** [comment]
Today I tried to use https again, I'm still observing the very same issue described on this thread. I'll switch back to old plain text then.

**Comment by madnight** [comment]
Same problem over here. I use https://github.com/worr/node-imdb-api and the omdb api requests sometimes are very slow for no obvious reason, especially if i request the episodes list for all seasons of a series.

**Comment by omdbapi** [comment]
I'm working with CloudFlare to get this resolved but need to know if you are experiencing a 524 or a 522 timeout error.

**Comment by josezulu** [comment]
I get Error 524: "Origin Time-out".

As mentioned in [issue 50](https://github.com/omdbapi/OMDb-API/issues/50), it seems to be directly related to run a search with the year parameter when using HTTPS.

**Comment by omdbapi** [comment]
This issue seems to be resolved after I activated Cloudflares Argo routing: https://www.cloudflare.com/products/argo-smart-routing/

**Which is a very expensive fix.** I'm investigating a few other options, one of which requires requests to be sent to a different subdomain to bypass Cloudflare entirely.

**Comment by josezulu** [comment]
I've seen 4 timeouts in about 20 requests in the span of 2 hours. Has the Argo service been turned off perhaps?

**Comment by omdbapi** [comment]
@josezulu You are correct, Argo is currently off while we try to track down the timeout issue. With it on everything works normally.
